src/foam/nanos/auth/Address.js

```diff
@@ -51,31 +51,20 @@ foam.CLASS({
     {
       class: 'String',
       name: 'address1',
-      // required: true
       width: 70,
       displayWidth: 50,
       documentation: 'An unstructured field for the main postal address.',
-      validateObj: function(address1) {
-        var address1Regex = /^[a-zA-Z0-9 ]{1,70}$/;
-
-        if ( address1.length > 0 && ! address1Regex.test(address1) ) {
-          return 'Invalid address line.';
-        }
-      }
+      expression: function(structured, streetNumber, streetName) {
+        return structured ? streetNumber + ' ' + streetName : '';
+      },
+      minLength: 1
     },
     {
       class: 'String',
       name: 'address2',
       width: 70,
       displayWidth: 50,
-      documentation: 'An unstructured field for the sub postal address.',
-      validateObj: function(address2) {
-        var address2Regex = /^[a-zA-Z0-9 ]{1,70}$/;
-
-        if ( address2.length > 0 && ! address2Regex.test(address2) ) {
-          return 'Invalid address line.';
-        }
-      }
+      documentation: 'An unstructured field for the sub postal address.'
```

Comment:
"sub postal address" is meaningless to me.

Reply:
Me too. Once this crunch is over I want to remove a ton of stuff from this model.

Reply:
I meant that the documentation string could maybe be updated, but that's minor lol.